Tasks/ServiceFabricDeploy/Create-DiffPackage.psm1

```diff
@@ -0,0 +1,150 @@
+{
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory=$true)][String] $ApplicationName,
+        [parameter(Mandatory=$true)][String] $ApplicationPackagePath,
+        [parameter(Mandatory=$true)] $ConnectedServiceEndpoint,
+        [parameter(Mandatory=$true)][Hashtable] $ClusterConnectionParameters
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try {
+        Write-Host (Get-VstsLocString -Key DIFFPKG_CreatingDiffPackage)
+
+        Import-Module $PSScriptRoot\ps_modules\ServiceFabricHelpers
+        Import-Module $PSScriptRoot\ps_modules\PowershellHelpers
+
+        $appManifestName = "ApplicationManifest.xml"
+        $localAppManifestPath = Join-Path $ApplicationPackagePath $appManifestName
+        $localAppManifestXml = [XML](Get-Content $localAppManifestPath)
+        $applicationTypeName = $localAppManifestXml.ApplicationManifest.ApplicationTypeName
+        $localAppTypeVersion = $localAppManifestXml.ApplicationManifest.ApplicationTypeVersion
+
+        $app = Get-ServiceFabricApplication -ApplicationName $ApplicationName
+
+        # If $app is null, it means the application does not exist in the cluster. Diff Package is equal to Full Package. Use Full Package to do deployment
+        if (!$app -or $app.ApplicationTypeName -ne $applicationTypeName)
+        {
+            Write-Host (Get-VstsLocString -Key DIFFPKG_ApplicationDoesNotExist -ArgumentList @($ApplicationName, $ConnectedServiceEndpoint.Url))
+            Return
+        }
+
+        $diffPackagePath = Join-Path (Get-VstsInput -Name diffPackagePath -Require) "DiffPackage"
+        if (Test-Path -PathType Container -Path $diffPackagePath)
+        {
+            Remove-Item -Path $diffPackagePath -Recurse -Force
+        }
+        $diffPackagePath = New-Item -ItemType Directory -Path $diffPackagePath -Force
+        $diffPkgAppManifestPath = Join-Path $diffPackagePath $appManifestName
+
+        $clusterAppTypeVersion = $app.ApplicationTypeVersion
+
+        # If the ApplicationTypeVersion of the Application is not upgraded, no diff package is made because there is no need
+        if ($clusterAppTypeVersion -eq $localAppTypeVersion)
+        {
+            Write-Host (Get-VstsLocString -Key DIFFPKG_ApplicationIsNotChanged -ArgumentList @($ApplicationName, $clusterAppTypeVersion, $ConnectedServiceEndpoint.Url))
+            Return
+        }
+
+        # Get the service types from the cluster
+        $serviceTypes = Get-ServiceFabricServiceType -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $clusterAppTypeVersion
+        # Pack the service manifest names into an array
+        $clusterServiceManifestNames = $serviceTypes.ServiceManifestName
+
+        # If $clusterServiceManifestNames is null, it means no service types are registered. Services can be registered without running but will still show up in this list in that case. 
+        # Diff Package is equal to Full Package. Use Full Package to do deployment
```

Comment:
The second part of my comment was just for you edification. I would leave the comment as:
\# If $clusterServiceManifestNames is null, it means no service types are registered. Diff Package is equal to Full Package. Use Full Package to do deployment